Add logging 2.1 to 2.2/3.0 migration guide (#10122)
aspnet/Docs: aspnetcore/migration/22-to-30.md

---
title: Migrate from ASP.NET Core 2.2 to 3.0
author: tdykstra
description: This article outlines the prerequisites and most common steps for migrating an ASP.NET Core 2.2 project to ASP.NET Core 3.0.
ms.author: tdykstra
ms.custom: mvc
ms.date: 01/04/2019
uid: migration/22-to-30
---
# Migrate from ASP.NET Core 2.2 to 3.0

By [Scott Addie](https://github.com/scottaddie)

> This article explains how to update an existing ASP.NET Core 2.2 project to ASP.NET Core 3.0 Preview.

Reviewer: ```suggestion
This article explains how to update an existing ASP.NET Core 2.2 project to ASP.NET Core 3.0.
```

[!INCLUDE[](~/includes/net-core-prereqs-all-3.0.md)]

## Update the project file

* Set the [Target Framework Moniker (TFM)](/dotnet/standard/frameworks#referring-to-frameworks) to `netcoreapp3.0`:

  ```xml
  <TargetFramework>netcoreapp3.0</TargetFramework>
  ```

> * Remove the `<PackageReference>` to `Microsoft.AspNetCore.All` and `Microsoft.AspNetCore.App`.

Reviewer: ```suggestion
* Remove any `<PackageReference>` to `Microsoft.AspNetCore.All` or `Microsoft.AspNetCore.App`.
```

> * Remove any `<PackageReference>` to [Microsoft.AspNetCore.All](xref:fundamentals/metapackage)` or [Microsoft.AspNetCore.App](xref:fundamentals/metapackage-app).

Reviewer 2: AFAIK (without much discussion on it), Rick, Scott, and I adopted a pattern while you were away. The links to the metapackage topics will say "metapackage" in the link, so I suggest (and there's a stray backtick there, too) ...

```suggestion
* Remove any `<PackageReference>` to [Microsoft.AspNetCore.All metapackage](xref:fundamentals/metapackage) or [Microsoft.AspNetCore.App metapackage](xref:fundamentals/metapackage-app).
```

>   **Preview 1 Known Issue** - workaround for [NuGet/Home #7342](https://github.com/NuGet/Home/issues/7342) - projects that do not start with `<Project Sdk="Microsoft.NET.Sdk.Web">` will get compiler or runtime errors due to missing Microsoft.AspNetCore.* assemblies. This is most often the case for test projects and class libraries. The workaround is to add the following to the *.csproj* file.

Reviewer: ```suggestion
  **Preview 1 Known Issue** - workaround for [NuGet/Home #7342](https://github.com/NuGet/Home/issues/7342) - projects that don't start with `<Project Sdk="Microsoft.NET.Sdk.Web">` will get compiler or runtime errors due to missing `Microsoft.AspNetCore.*` assemblies. This is most often the case for test projects and class libraries. The workaround is to add the following to the *.csproj* file.
```

>   **Preview 1 Known Issue** - workaround for [NuGet/Home #7342](https://github.com/NuGet/Home/issues/7342) - projects that don't start with `<Project Sdk="Microsoft.NET.Sdk.Web">` will get compiler or runtime errors due to missing `Microsoft.AspNetCore.*` assemblies. This is most often the case for test projects and class libraries. The workaround is to add the following to the *.csproj* file.

Reviewer 2: *ugh* ... imo, this doesn't look good and the future tense isn't necessary ... but ok.

  ```xml
  <ItemGroup>
     <FrameworkReference Include="Microsoft.AspNetCore.App" />
  </ItemGroup>
  ```

* Update the `Version` on remaining `<PackageReference>` elements for `Microsoft.AspNetCore.*` packages to 3.0.0-preview-18579-0053 (or whatever the number is for the current Preview)

>   If there is no 3.0 version of a package, the package might have been deprecated in 3.0. Many of these are part of `Microsoft.AspNetCore.App` and don't need to be referenced individually anymore. For a preliminary list of packages no longer produced in 3.0, see https://github.com/aspnet/AspNetCore/issues/3756.

Reviewer 3: > Is there a list of deprecated packages in 3.0?

https://github.com/aspnet/AspNetCore/issues/3756

>   If there is no 3.0 version of a package, the package might have been deprecated in 3.0. Many of these are part of `Microsoft.AspNetCore.App` and should not be referenced individually anymore. For a preliminary list of packages no longer produced in 3.0, see [aspnet/AspNetCore #3756](https://github.com/aspnet/AspNetCore/issues/3756).

Reviewer: ```suggestion
  If there is no 3.0 version of a package, the package might have been deprecated in 3.0. Many of these are part of `Microsoft.AspNetCore.App` and shouldn't be referenced individually anymore. For a preliminary list of packages no longer produced in 3.0, see [aspnet/AspNetCore #3756](https://github.com/aspnet/AspNetCore/issues/3756).
```

> * Some APIs were removed from Microsoft.AspNetCore.App between 2.x and 3.0. You may need to add `<PackageReference>` items if you're using API from packages listed in https://github.com/aspnet/AspNetCore/issues/3755

Reviewer 2: I always include the title in these types of issue links. Suggestion (this is what I use) ...

> \[Assemblies being removed from Microsoft.AspNetCore.App 3.0 (aspnet/AspNetCore #3755)](https:\/\/github.com/aspnet/AspNetCore/issues/3755)

> * Some assemblies were removed from Microsoft.AspNetCore.App between 2.x and 3.0. You may need to add `<PackageReference>` items if you're using APIs from packages listed in [aspnet/AspNetCore #3755](https://github.com/aspnet/AspNetCore/issues/3755)

Reviewer: ```suggestion
* Some assemblies were removed from `Microsoft.AspNetCore.App` between 2.x and 3.0. You may need to add `<PackageReference>` items if you're using APIs from packages listed in [aspnet/AspNetCore #3755](https://github.com/aspnet/AspNetCore/issues/3755)
```

>   For example, Entity Framework Core and System.Data.SqlClient are no longer part of Microsoft.AspNetCore.App.

Reviewer 4: @tdykstra 	can you review my suggestions and apply as appropriate?

>   For example, Entity Framework Core and System.Data.SqlClient are no longer part of Microsoft.AspNetCore.App.

Reviewer 2: ... and we pretty much always link to our metapackage topic (with the word "metapackage" in the link) ...

> \[Microsoft.AspNetCore.App metapackage](xref:fundamentals/metapackage-app)

>   For example, `Microsoft.EntityFrameworkCore` and `System.Data.SqlClient` are no longer part of Microsoft.AspNetCore.App.

Reviewer: ```suggestion
  For example, `Microsoft.EntityFrameworkCore` and `System.Data.SqlClient` are no longer part of `Microsoft.AspNetCore.App`.
```

>   **Preview 1 Known Issue** - the list of assemblies shipping in `Microsoft.AspNetCore.App` has not been finalized yet and will change before 3.0 RTM.

Reviewer 2: I'd WARNING this one, too.

>   **Preview 1 Known Issue** - the list of assemblies shipping in `Microsoft.AspNetCore.App` has not been finalized yet and will change before 3.0 RTM.

Reviewer: ```suggestion
  **Preview 1 Known Issue** - the list of assemblies shipping in `Microsoft.AspNetCore.App` hasn't been finalized yet and will change before 3.0 RTM.
```

>   **Preview 1 Known Issue** - the list of assemblies shipping in `Microsoft.AspNetCore.App` hasn't been finalized yet and will change before 3.0 RTM.

Reviewer 2: Same here.
